### PR TITLE
Voice Recorder

### PR DESCRIPTION
**Pull request by BombasterDS**
## About the PR
Added Voice Recorder that can record conversation between people!
You can start and stop recording with 5 cd, then print transcript of your conversation.
Have 4 listening radius.
Can we found in detective's locker and security vendmachine

## Why / Balance
Cool ss13 thing

## Technical details
When IsRecording entity get ActiveListenerComp that allows it to receive chat messages. All messages are stored in string array in VoiceRecorderComponent.

Adds two verbs for Record Cleaning (ActivateVerb) and Record Printing (AlternativeVerb). They won't appear if string array is empty or IsRecording true.

When Print it makes one Paper Entity that keep all messages

## Media

https://github.com/space-wizards/space-station-14/assets/115770678/84264fc4-8ba6-4af8-8685-b539a57b612f

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Voice Recorder that can record your conversation with friends!

### CODE REVIEW

**Comment by BombasterDS** [comment]
No way it worked without test failure!

**Comment by Brandon-Huu** [comment]
> No way it worked without test failure

Why did you make it print paper instead of playing back what was said?

**Comment by BombasterDS** [comment]
> > No way it worked without test failure
> 
> Why did you make it print paper instead of playing back what was said?

Imagine listening 10+ minutes conversation without ability to pause or rewind playback 💀💀💀
It's just easier and more flexible in use

**Review by ficcialfaint — CHANGES_REQUESTED** [review]
New audio files should be stated in `attributions.yml`, see https://docs.spacestation14.com/en/specifications/robust-generic-attribution.html

**Comment by Brandon-Huu** [comment]
> 10+ minutes conversation without ability to pause or rewind playback

Can you not just give the device a UI for this?

**Comment by beck-thompson** [comment]
Great idea! This will be really fun for lawyers.
I did do some tests and it doesn't seem to work with the voice mask system so you should probably check that out! Sec will 100% metagame it the recorder is easy to obtain.

It would also be nice to have a verb that lets you start / stop recording but if you only want it to work in hand that also makes sense.

**Comment by BombasterDS** [comment]
> Great idea! This will be really fun for lawyers. I did do some tests and it doesn't seem to work with the voice mask system so you should probably check that out! Sec will 100% metagame it the recorder is easy to obtain.
> 
> It would also be nice to have a verb that lets you start / stop recording but if you only want it to work in hand that also makes sense.

Fixed voice mask. About verbs - i don't think that it need specific verb for turn on/off. You can just activate by pressing E in your inventory or in world,

File: Resources/Audio/Items/Recorder/attributions.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- files: ["recorder_erase.ogg", "recorder_start.ogg", "recorder_stop.ogg"]
+  license: "СС-BY-NC-SA-3.0"
+  copyright: "Taken from Zvukipro.com"
```
Comment (metalgearsloth) [inline Resources/Audio/Items/Recorder/attributions.yml:3]: Can you point out where the license is listed on the site please.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
1. None of the code is documented.
2. All of it should be in shared.

Will review the rest from there.

Also is there a reason you didn't use one of the old PRs for this.

File: Resources/Audio/Items/Recorder/attributions.yml (hunk 1 above)
Comment (BombasterDS) [reply Resources/Audio/Items/Recorder/attributions.yml]: > Can you point out where the license is listed on the site please.

I didn't found exact line with license name, but it says that is free to use if you stay the site link (CC-BY-SA as i know right) - https://zvukipro.com/index.php?do=feedback

Comment (BombasterDS) [reply Resources/Audio/Items/Recorder/attributions.yml]: Don't really understand much about licenses

Comment (BombasterDS) [reply Resources/Audio/Items/Recorder/attributions.yml]: About old PR, i didn't saw them earlier :|. I watched them now and think, that they look much better then mine. I'll try to ask authors about them

**Comment by Cojoke-dot** [comment]
Add it to the law drobe, lawyers would want this.

File: Resources/Audio/Items/Recorder/attributions.yml (hunk 1 above)
Comment (Hrosts) [reply Resources/Audio/Items/Recorder/attributions.yml]: They don't specify a license there and they also host the obviously licensed sounds from AAA games (Zelda BotW, Prey, Bioshock Infinite). It's almost guaranteed the sound you've used is owned by someone.

The primary source for the station's sound effects is https://freesound.org/, it seems, while [this search](https://github.com/search?q=repo%3Aspace-wizards%2Fspace-station-14+.mp3+OR+.ogg+OR+.m4a+OR+.flac+path%3A**%2Fattributions.txt&type=code) will show you the attributions files and the sources for the specific sounds.

**Comment by lapatison** [comment]
Add it to the reporter as well, please

**Comment by Slardar0609** [comment]
Man, I had such high hopes for this. It's an incredible PR, but the fact that it prints paper instead of playing back is utterly depressing for me. No jigsaw tapes for my traps :<

**Comment by Brandon-Huu** [comment]
> Man, I had such high hopes for this. It's an incredible PR, but the fact that it prints paper instead of playing back is utterly depressing for me. No jigsaw tapes for my traps :<

You could always make a PR to make it play back audio instead.

> Imagine listening 10+ minutes conversation without ability to pause or rewind playback

Could just make a UI similar to how midis and juke boxes work to solve this.

**Comment by BombasterDS** [comment]
https://github.com/space-wizards/space-station-14/pull/30089
↑↑↑

**Pull request closed without merging**